File: App/Backend/Models/FullPost.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Backend.Models
{
    public class FullPost
    {
       
        public long postId { get; set; }
        public long userId { get; set; }
        public string username { get; set; }
        public long postTypeId { get; set; }
        public string typeName { get; set; }
        public DateTime createdAt { get; set; }
        public string description { get; set; }
        public string photoPath { get; set; }
        public long statusId { get; set; }
        public string status { get; set; }
        public int likeNum { get; set; }
        public int dislikeNum { get; set; }

        public FullPost(Post p) //post se posalje
        {
            this.username = p.user.username;
            this.createdAt = p.createdAt;
            this.description = p.description;
            this.dislikeNum = p.likes.Count(x => x.likeTypeId == 1);
            this.likeNum = p.likes.Count(x => x.likeTypeId == 2);
            this.photoPath = p.photoPath;
            this.postId = p.id;
            this.postTypeId = p.postTypeId;
            this.status = p.status.type;
            this.statusId = p.statusId;
            this.userId = p.userId;
            this.typeName = p.postType.typeName;
          
            
           
        }


    }
}

File: App/Frontend/lib/models/fullPost.dart
class FullPost
{
   
  int _postId;
  int _userId; 
  String _username;
  int _postTypeId;
  String _typeName;
  DateTime _createAt; //time the post was posted
  String _description;
  String _photoPath; 
  int _statusId;
  String _status; // 'reseno' or 'nereseno'
  int _likeNum; //number of likes
  int _dislikeNum; //number of dislikes
  int _commNum;
  
  FullPost(this._postId, this._userId, this._username,this._postTypeId,this._typeName,this._createAt,this._description,this._photoPath,this._statusId,this._status,this._likeNum,this._dislikeNum, this._commNum );
 

  int get postId => _postId;
  int get userId => _userId;
  String get username => _username;
  int get postTypeId => _postTypeId;
  String get typeName => _typeName;
  DateTime get createAt => _createAt;
  String get description => _description;
  String get photoPath => _photoPath;
  int get statusId => _statusId;
  String get status => _status;
  int get likeNum => _likeNum;
  int get dislikeNum => _dislikeNum;
  int get commNum => _commNum;
  

 //saljemo kao json fajl
  Map<String, dynamic> toMap()
  {
    var map = Map<String, dynamic>();
    
    map["postId"] = _postId;
    map["userId"] = _userId;
    map["username"] = _username;
    map["postTypeId"] = _postTypeId;
    map["typeName"] = _typeName;
    map["createAt"] = _createAt;
    map["description"] = _description;
    map["photoPath"] = _photoPath;
    map["statusId"] = _statusId;
    map["status"] = _status;
    map["likeNum"] = _likeNum;
    map["dislikeNum"] = _dislikeNum;
    map["commNum"] = _commNum;
    
   
    return map;
  }

  //pretvaramo iz json-a u object 
  FullPost.fromObject(dynamic data)
  {
    this._postId = data["postId"];
    this._userId = data["userId"];
    this._username = data["username"];
    this._postTypeId = data["postTypeId"];
    this._typeName = data["typeName"];
    this._createAt = data["createAt"];
    this._description = data["description"] ;
    this._photoPath = data["photoPath"];
    this._statusId = data["statusId"];
    this._status = data["status"];
    this._likeNum = data["likeNum"];
    this._dislikeNum = data["dislikeNum"];
    this._commNum = data["commNum"];
    

  }
}